Go/sereal/sereal_test.go

```diff
@@ -68,14 +70,42 @@ func testRoundtrip(t *testing.T, perlCompat bool, version int) {
 			t.Errorf("failed marshalling with perlCompat=%t : %v: %s\n", perlCompat, v, err)
 		}
 		var unp interface{}
-
 		err = d.Unmarshal(b, &unp)
 		if err != nil {
-			t.Errorf("perl compat: error during unmarshall: %s\n", err)
+			t.Errorf("error during unmarshal: %s\n", err)
 		}
+
 		if !reflect.DeepEqual(v, unp) {
 			t.Errorf("failed roundtripping with perlCompat=%t: %#v: got %#v\n", perlCompat, v, unp)
 		}
+
+		// for a couple of common "root" types also try unmarshalling to the same type
+		switch vt := v.(type) {
+		default:
+			_ = vt // avoid "declared but not used"
+
+		case []interface{}:
+			unSlice := make([]interface{}, 0)
```

Comment:
Can this (and the one below) changed to:

```
var unSlice []interface{}
```

or is the initialization necessary for the test to be meaningful?

Reply:
No, it is not necessary. Will update.